tests/MusicTests.elm

```diff
@@ -0,0 +1,205 @@
+                , bNatural four (eighthNote |> tied)
+                , bNatural four halfNote
+                , cSharp five eighthNote
+                , bNatural five eighthNote
+                ]
+
+        --bars21-23
+        v2f =
+            line
+                [ fSharp five eighthNote
+                , aNatural five eighthNote
+                , bNatural five (halfNote |> tied)
+                , bNatural five quarterNote
+                , aNatural five eighthNote
+                , fSharp five eighthNote
+                , eNatural five quarterNote
+                , dNatural five eighthNote
+                , fSharp five eighthNote
+                , eNatural five halfNote
+                , dNatural five halfNote
+                , fSharp five quarterNote
+                ]
+
+        -- bars 24-28
+        v2g =
+            line
+                [ eighthNoteTriplet (cSharp five) (dNatural five) (cSharp five)
+                , line
+                    [ bNatural four (halfNote |> tied)
+                    , bNatural four (dotted halfNote |> tied)
+                    , bNatural four (dotted halfNote |> tied)
+                    , bNatural four (dotted halfNote)
+                    ]
+                ]
+
+        v2 =
+            line [ v2a, v2b, v2c, v2d, v2e, v2f, v2g ]
+
+        mainVoice =
+            seq (times 3 v1) v2
+    in
+    tempo (dotted halfNote) 69 <|
+        key
+            (PitchClass.pitchClass E natural |> Key.major)
+            (timeSignature 3 4 (par bassLine mainVoice))
+
+
+all : Test
+all =
+    describe "Music Tests"
+        [ test "get all pitches from a music" <|
+            \_ ->
+                twoFiveOne
+                    |> map Spelling.simple
+                    |> toList
+                    |> Expect.equal
+                        [ Ok { letter = D, accidental = Natural, octave = four }
+                        , Ok { letter = F, accidental = Natural, octave = four }
+                        , Ok { letter = A, accidental = Natural, octave = four }
+                        , Ok { letter = D, accidental = Natural, octave = four }
+                        , Ok { letter = G, accidental = Natural, octave = four }
+                        , Ok { letter = B, accidental = Natural, octave = four }
+                        , Ok { letter = C, accidental = Natural, octave = four }
+                        , Ok { letter = E, accidental = Natural, octave = four }
+                        , Ok { letter = G, accidental = Natural, octave = four }
+                        ]
+        , test "children songs 6" <|
+            \_ ->
+                childrenSongNumber6
+                    |> toList
+                    |> List.length
+                    |> Expect.greaterThan 20
+        , test "create tuplets" <|
+            \_ ->
+                quarterNoteTriplet (cNatural four) (chord [ cNatural four, eNatural four ]) (chord [ cNatural four, eNatural four, gNatural four ])
```

Comment:
This is an example of creating an eighth note tuplet with single notes and chords mixed. This has the guaranty that the tuplet durations are correct.

Reply:
Note that this could by bypassed like this

    quarterNoteTriplet (\_ -> cNatural four halfNote) ... etc.